plugins/contactpressuresensorsarray/src/ContactPressureSensorsArrayDriver.cpp

```diff
@@ -0,0 +1,516 @@
+/*
+ * Copyright (C) 2018 Fondazione Istituto Italiano di Tecnologia RBCS & iCub Facility & ADVR
+ * Authors: see AUTHORS file.
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or any later version, see LGPL.TXT or LGPL3.TXT
+ */
+
+#include "ContactPressureSensorsArrayDriver.h"
+#include <GazeboYarpPlugins/Handler.hh>
+#include <GazeboYarpPlugins/common.h>
+
+#include <yarp/os/LockGuard.h>
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/math/Math.h>
+#include <yarp/math/SVD.h>
+
+#include <sdf/Param.hh>
+
+#define DEBUG 0
+#define N_AXIS 3
+#define TIME_CONVERSION_NS 1.0e-9
+
+using namespace yarp::dev;
+
+GazeboYarpContactPressureSensorsArrayDriver::GazeboYarpContactPressureSensorsArrayDriver()
+{
+
+}
+
+GazeboYarpContactPressureSensorsArrayDriver::~GazeboYarpContactPressureSensorsArrayDriver()
+{
+
+}
+
+void GazeboYarpContactPressureSensorsArrayDriver::onUpdate(const gazebo::common::UpdateInfo& _info)
+{ 
+    gazebo::msgs::Contacts contacts;
+    contacts = this->m_sensor->Contacts();
+    
+    ignition::math::Vector3d resultantForceonCG;
+    ignition::math::Vector3d resultantMomentonCG;
+        
+    double timestamp = -1;
+    for (size_t i = 0; i < contacts.contact_size(); i++)
+    {
+        gazebo::common::Time currentStamp;
+        for (int j=0; j < contacts.contact(i).position_size(); j++)
+        {
+            double t = contacts.contact(i).time().sec() + TIME_CONVERSION_NS*contacts.contact(i).time().nsec();
```

Comment:
I think this line can be simplified to:
~~~
double t = contacts.contact(i).time.Double() 
~~~

Reply:
For some reason, `time` is pointing towards `gazebo::msgs::Time` instead of `gazebo::common::Time` and makes it unable to access the `Double` method of the `Time` class. So, sticking to the lengthy version.

Reply:
I see, ok.